### DISCUSSION TITLE
how to get actual exception from context

### DISCUSSION THREAD

**Discussion by Alnyz** [apache/airflow#29717] [Q&A]
i want to see actual exception error from my proggram, but i cant found any in `context` after using `on_failure_callback`

my dag code:
```python

def when_fail(context):
    print(context)
    
with DAG(
    dag_id="testv1",
    start_date=datetime(2018, 1, 30),
    catchup=False,
) as dag:
    task1 = BashOperator(
        task_id='test1',
        dag=dag,
        owner='airflow',
        on_failure_callback=when_fail,
        bash_command='python3 /mnt/c/dags/test.py',
      )
    task1
```

and code in `test.py`

```python
1/0
```
that code should raise error `ZeroDivisionError: division by zero`, and this error what i want to catch.
i try using `context['exception']` this throwing `airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.`

**Comment by Taragolis** [comment]
There is no exceptions at all in `bash` as well as other unix/posix shells.

**Reply by potiuk** [reply]
BTW. You should use PythonOperator if you want the exceptions to propagate. Having Bash in-between Airflow (Python) and your python code, you are loosing the exception information as @Taragolis mentioned. Not sure what details you will find in the exception, when you use PythonOperator, but you are increasing your chances of getting a meaningful one.
